debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerStreamingChangeEventSource.java

```diff
@@ -105,7 +107,7 @@ public SqlServerStreamingChangeEventSource(SqlServerConnectorConfig connectorCon
                         ? DEFAULT_INTERVAL_BETWEEN_COMMITS.toMillis()
                         : intervalBetweenCommitsBasedOnPoll.toMillis());
         this.pauseBetweenCommits.hasElapsed();
-        this.streamingExecutionContexts = new HashMap<>();
+        this.streamingExecutionContexts = Collections.synchronizedMap(new HashMap<>());
```

Comment:
Let's make it a separate Jira case: refactor this to avoid the need for synchronization by moving the newly added properties out of the streaming context.